tools/src/main/scala/org/apache/spark/sql/rapids/tool/profiling/ApplicationInfo.scala

```diff
@@ -203,30 +207,62 @@ class ApplicationInfo(
   // Create Spark DataFrame(s) based on ArrayBuffer(s)
   arraybufferToDF()
 
+  private val codecMap = new ConcurrentHashMap[String, CompressionCodec]()
+
+  def openEventLogInternal(log: Path, fs: FileSystem): InputStream = {
+    EventLogFileWriter.codecName(log) match {
+      case c if (c.isDefined && c.get.equals("gz")) =>
+        val in = new BufferedInputStream(fs.open(log))
+        try {
+          new GZIPInputStream(in)
+        } catch {
+          case e: Throwable =>
+            in.close()
+            throw e
+        }
+      case _ => EventLogFileReader.openEventLog(log, fs)
+    }
+  }
+
   /**
    * Functions to process all the events
    */
   def processEvents(): Unit = {
-    logInfo("Parsing Event Log File: " + eventlog.toString)
+    val eventlog = eventLogInfo.eventLog
 
-    val fs = FileSystem.get(eventlog.toUri,new Configuration())
-    var totalNumEvents = 0
+    logInfo("Parsing Event Log: " + eventlog.toString)
 
+    // at this point all paths should be valid event logs or event log dirs
+    val fs = eventlog.getFileSystem(new Configuration())
```

Comment:
prefer to wire through hadoop conf from SparkContext to creating brand-new instances.